src/naga.cpp

```diff
@@ -15,6 +15,7 @@
 #include <mutex>
 #include <map>
 #include <sstream>
+#include <algorithm>
```

Comment:
I think this is unsafe. The error must be corrected with the right version of g++ or the right cast

Reply:
Sorry man, i am not really a c++ programmer, just wanted to help..
Please just add the naga-x path to your repo, I'll delete the fork.